modules/reindex/src/test/java/org/elasticsearch/index/reindex/ReindexFromRemoteBuildRestClientTests.java

```diff
@@ -49,4 +49,29 @@ public void testBuildRestClient() throws Exception {
             client.close();
         }
     }
+    
+    public void testBuildRestClientWithHeader() throws Exception {
+        Map<String, String> headers = new HashMap<>();
+        int headerNumbers = randomIntBetween(2, 5);
+        for (int i = 0; i < headerNumbers; i++) {
+            headers.put("header" + i, Integer.toString(i));
+        }
+        assertEquals(headerNumbers, headers.size());
+        System.out.println(headers.toString());
+        RemoteInfo remoteInfo = new RemoteInfo("https", "localhost", 9200, new BytesArray("ignored"), null, null, headers,
+            RemoteInfo.DEFAULT_SOCKET_TIMEOUT, RemoteInfo.DEFAULT_CONNECT_TIMEOUT);
+        long taskId = randomLong();
+        List<Thread> threads = synchronizedList(new ArrayList<>());
+        RestClient client = TransportReindexAction.buildRestClient(remoteInfo, taskId, threads);
```

Comment:
I agree with @jasontedor. I'd be fine with a adding `RestClient#getDefaultHeaders` which returns a copy of the `defaultHeaders` array so we can assert that they are correct. @javanna, what do you think of that?

Also, I don't think there is any need to assert on the threads in this test. The test above is careful about that already.